build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -1600,6 +1662,78 @@ message BatchUpdateBlobsResponse {
   repeated Response responses = 1;
 }
 
+message ConcatenateBlobsRequest {
+  message Request {
+    // The digest of the blob to create.
+    Digest digest = 1;
+
+    // For BLAKE3CONCAT, the message words m0...m15 of the root parent
+    // nodes of the smaller blobs, stored in little-endian order.
+    //
+    // The first 64*n bytes correspond to the root nodes of n blobs
+    // having size 2^k, where k > 10. The final 64 bytes correspond to
+    // the root node of the blob containing trailing data, containing l
+    // bytes of data, where 0 < l <= 2^k. The server may infer the
+    // values of k and l using `digest.size_bytes` and the length of
+    // this field.
+    bytes chaining_values = 2;
```

Comment:
So initially I actually had a `repeated fixed32` here, so that the 32-bit integers would be stored as little-endian values directly. The reason I picked `bytes` here is to keep this generic enough, so that it may work for other hashing algorithms as well.

The documentation that I gave above purely reflects how this field is used in the case of BLAKE3CONCAT. Other hashing algorithms are free to use other layouts/representations that are not necessarily record based.

Reply:
No longer applies, as we now use string hashes here.